### ISSUE TITLE
Non-integer values in EventArray.arrayValues causing crash

### ISSUE THREAD

**Issue by CFlaniganMide** [MideTechnology/idelib#12]
Non-integer start and end values in `EventArray.arrayValues()` is causing a crash.  There are likely more methods affected by this.

**Comment by CrepeGoat** [comment]
Obviously I'm late to this thread, but for documentation, why should the array-slice methods support floating-point ranges?

**Comment by StokesMIDE** [comment]
Floats can end up getting used if the number got implicitly cast to float during calculation. I forget what specifically prompted this issue, but that seems like the most likely case in which this would occur.

**Comment by CrepeGoat** [comment]
Let me rephrase my question: numpy arrays don't support float indices. lists don't support float indices. Why should `EventArray` support float indices?

**Comment by CFlaniganMide** [comment]
good question, and one I sadly don't remember the answer to, even after looking at the old code.

**Comment by CrepeGoat** [comment]
@CFlaniganMide I appreciate the honest answer.

This probably goes without saying, but if I had the choice at the time I would rather this change *not* have happened, for a handful of reasons:
1) for consistency with lists and numpy arrays,
2) so that users (which we have now I guess) don't mis-assume that they're indexing on *timestamps* instead of integer sample numbers,
and 3) because any prior errors that prompted this change originally may be indicative of something wrong in the corresponding index calculations, which should be fixed but are now hidden. @CFlaniganMide do you remember in what code you were seeing errors? (Lab, cloud, some other script, etc.)

**Comment by CFlaniganMide** [comment]
1. There's plans to make them compatible, as part of the NRE project category.  I'll be working on that shortly after tow tester wraps (hopefully before those plans have time to change).
2. Agreed, timestamps shouldn't use the bracket indexing format.  Hopefully if we can get everything running as a numpy array, then we can just toss it at a numpy interpolate routine. 
3. It's true, but like I said, not sure what the cause is.  Setting warnings before reverting changes might help us locate issues before we start working on them.

**Comment by CrepeGoat** [comment]
A fix for this was already merged in -> this should be ok for closing. I'll add my concerns in a separate item.
